app/components/chronos-submission-panel/component.js

```diff
@@ -0,0 +1,74 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
+import { A } from '@ember/array';
+
+export default Component.extend({
+    store: service(),
+    toast: service(),
+    isLoading: false,
+    canLoadMore: false,
+    page: 1,
+    selectedJournal: null,
+    keyword: '',
+    preprint: null,
+    journals: A(),
+    _fetchInitialJournals: task(function* (keyword) {
+        // Wait a bit for the user to finish typing.
+        yield timeout(500);
+        // Reset the list of journals
+        this.set('journals', A());
+        // Reset page number
+        this.set('page', 1);
+        this.set('isLoading', true);
+        const journals = yield this.get('store').query('chronos-journal', { page: 1, 'filter[title]': keyword });
+        this.set('isLoading', false);
+        const canLoadMore = !(journals.get('meta.total_pages') === this.get('page'));
+        // Set the keyword
+        this.set('keyword', keyword);
+        this.set('canLoadMore', canLoadMore);
+        this.set('journals', journals);
+    }).restartable(),
+    actions: {
```

Comment:
do these need analytics? Question for demo

Reply:
Could go into an improvement ticket.